fe/fe-core/src/main/java/com/starrocks/catalog/PhysicalPartition.java

```diff
@@ -500,4 +493,59 @@ public void gsonPostProcess() throws IOException {
             versionTxnType = TransactionType.TXN_NORMAL;
         }
     }
+
+    // partition data size reported by be, but may be not accurate
+    public long getDataSize() {
+        long dataSize = 0;
+        for (MaterializedIndex mIndex : getMaterializedIndices(IndexExtState.VISIBLE)) {
+            dataSize += mIndex.getDataSize();
+        }
+        return dataSize;
+    }
+
+    // partition storage size reported by be, but may be not accurate
+    public long getStorageSize() {
+        long storageSize = 0;
+        for (MaterializedIndex mIndex : getMaterializedIndices(IndexExtState.VISIBLE)) {
+            storageSize += mIndex.getStorageSize();
+        }
+        return storageSize;
+    }
+
+    public long getLastSuccVacuumTime() {
+        return lastSuccVacuumTime;
+    }
+
+    public void setLastSuccVacuumTime(long lastVacuumTime) {
+        this.lastSuccVacuumTime = lastVacuumTime;
+    }
+
+    public boolean shouldVacuum() {
```

Comment:
Don't put external logic in the definition of the Partition class, put it where it is used